core/trino-spi/src/main/java/io/trino/spi/predicate/Utils.java

```diff
@@ -64,4 +72,53 @@ static RuntimeException handleThrowable(Throwable throwable)
         }
         return new RuntimeException(throwable);
     }
+
+    /**
+     * Try to expand {@code valueSet} into a discrete set of (at most {@code limit}) objects.
+     * For example: [1, 5] can be expanded into {1, 2, 3, 4, 5}.
+     * If the data type is not supported or the expansion results in too many values, {@code Optional.empty()} is returned.
+     */
+    public static Optional<Collection<Object>> tryExpandRanges(ValueSet valuesSet, int valuesLimit)
+    {
+        if (!(valuesSet instanceof SortedRangeSet) || valuesLimit <= 0) {
```

Comment:
A negative limit should result in an error. There's no legitimate case where such a value should be provided.

Reply:
You're right - updated the code.